packages/manager/modules/telecom-spare/src/modems/modal/modems-modal.controller.js

```diff
@@ -0,0 +1,144 @@
+export default class {
+  /* @ngInject */
+  constructor($uibModalInstance, OvhApiXdsl, params) {
+    this.$uibModalInstance = $uibModalInstance;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.action = params.action;
+    this.spare = params.spare;
+  }
+
+  $onInit() {
+    this.isToReplace = false;
+    this.isToReturn = false;
+    this.isToDelete = false;
+    this.isApplyAvailable = false;
+
+    this.selectedDomain = '';
+
+    this.loading = false;
+    switch (this.action) {
+      case 'replacement':
+        this.isToReplace = true;
+        this.title = 'xdsl_modem_modal_replacement_title';
+        this.loading = true;
+        this.retrieveCompatibleReplacement();
+        break;
+      case 'return':
+        this.isToReturn = true;
+        this.title = 'xdsl_modem_modal_return_title';
+        this.isApplyAvailable = true;
+        break;
+      case 'delete':
+        this.isToDelete = true;
+        this.title = 'xdsl_modem_modal_delete_title';
+        this.isApplyAvailable = true;
+        break;
+      default:
+        break;
+    }
+  }
+
+  retrieveCompatibleReplacement() {
+    return this.OvhApiXdsl.Spare().v6().queryCompatibleReplacement({
+      spare: this.spare,
+    }).$promise
+      .then((result) => {
+        this.isApplyAvailable = true;
+        this.compatibleDomains = result;
+      })
+      .catch(() => {
+        this.isApplyAvailable = false;
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  apply() {
+    switch (this.action) {
+      case 'replacement':
+        this.replaceSpare(this.spare, this.selectedDomain);
+        break;
+      case 'return':
+        this.returnMerchandise(this.spare);
+        break;
+      case 'delete':
+        this.deleteSpare(this.spare);
```

Comment:
Should be 3 different components if 3 different behaviour ? Or one component but 3 different routing and `resolve` values